File: MyProject/myproject/static/js3 — копия/Motion.js
 var Motion = function(motion_vector, move_power, motion_color) {
  this.motion_vector = motion_vector;
  this.move_power = move_power;
  this.motion_color = motion_color;
}

var Motion = function(motion) {
  this.motion_vector = motion.motion_vector;
  this.move_power = motion.move_power;
  this.motion_color = motion.motion_color;
}

var Motion = function() {
  this.motion_vector = new Vector2(0,0);
  this.move_power = 0;
  this.motion_color = new Color(0,0,0,0);
}

Motion.prototype.SetAll = function(c) {
  this.motion_vector.SetAll(c.motion_vector);
  this.move_power = c.move_power;
  this.motion_color.SetAll(c.motion_color);
}

File: MySocketExp/game/Motion.py
import json


class Motion:
    __slots__ = ['motion_vector', 'move_power', 'motion_color']

    def __init__(self, motion_vector, move_power, motion_color):
        self.motion_vector = motion_vector
        self.motion_color = motion_color
        self.move_power = move_power

    def inc_motion_vector(self, add_position):
        self.motion_vector = add_position * self.move_power  # .normalize()


    def encode_json(self):
        return {
            'move_power': json.dumps(self.move_power),
            'motion_vector': self.motion_vector.encode_json(),
            'motion_color': self.motion_color.encode_json()
        }